processor/src/main/java/com/infinum/jsonapix/processor/specs/RelationshipsSpecBuilder.kt

```diff
@@ -90,6 +102,27 @@ internal object RelationshipsSpecBuilder {
             .build()
     }
 
+    private fun PropertySpec.Builder.addSerialNameAnnotationIfItExits(originalProperty: PropertySpec): PropertySpec.Builder = this.apply {
+        originalProperty.annotations.findAnnotationWithTypeName(serialNameTypeName)?.let { annotationSpec ->
+            addAnnotation(annotationSpec)
+        }
+    }
+
+    private fun mapPropertiesToParams(properties: List<PropertySpec>): List<ParameterSpec> {
+        return properties.map {
+            ParameterSpec.builder(it.name, it.type)
+                .apply {
+                    if (it.annotations.missingTypeName(serialNameTypeName)) {
+                        addAnnotation(Specs.getSerialNameSpec(it.name))
```

Comment:
I think we can combine the logic here with the logic in `addSerialNameAnnotationIfItExits` so we can have one single place to add the `@SerialName` annotation, Or maybe I'm missing something

Reply:
Hmm yes this is a bit confusing, I was copying the `AttributesSpecBuilder`, there we are always adding the `@SerialName` annotation, if it was **not defined** in the class we are serialising:

![Screenshot 2024-01-11 at 15 58 18](https://github.com/infinum/kotlin-jsonapix/assets/25981828/20f58158-4e3d-4c3b-a68f-d01a1c3955bd)

So the generated classes look like this:

![Screenshot 2024-01-11 at 16 00 55](https://github.com/infinum/kotlin-jsonapix/assets/25981828/424e09c2-84d3-4a4f-9d80-4104243e3b83)

The @SerialName value is the same as the variable name. So hmm maybe this is not needed? I do not know if removing this would break anything. 

I commented out the code and it seems fine(can serialise the class normally). But maybe you can check it a bit more😅 
```
if (it.annotations.missingTypeName(serialNameTypeName)) {
    addAnnotation(Specs.getSerialNameSpec(it.name))
}
```

So currently our flow is a bit weird. We check if the annotation is present in the original class and add it to the new `PropertySpec` which we are creating. Then later we check if this new `PropertySpec` has the annotation, and if it **does not**, we add the @SerialName annotation with the name of the variable.

Reply:
I think a concise solution here, is we create a new annotation every time regardless, and we get the value of the original annotation if it exists if not we can just use the key name. 

Even if it's not important if the value of @SerailName is the same as the key, however it makes the code more consistent

Reply:
This means from one single place, we can check the two situations if the serial name is used we go with it's value if not we just use the key instead